openassessment/xblock/submission_mixin.py

```diff
@@ -490,6 +490,44 @@ def get_files_info_from_user_state(self, username):
                     continue
         return files_info
 
+    def get_all_upload_urls(self, username):
+        """
+        For a particular ORA block, get the download URLs for all the files uploaded and still present.
+
+        Used for an extremely edge case, where the stored files indices are out of sync with
+        the uploaded files, this is a last resort to get the download URLs of all the files
+        that have been uploaded by a learner in an ORA block(and haven't been deleted from the storage).
+        Starting from 0 index to maximum file upload count possible, this checks if a file exists against
+        every index. If present, add the info, else repeat it for the next indices.
+
+        Arguments:
+            username(str): username or email of the learner whose files' information is to be obtained.
```

Comment:
`username` can this argument contain email or username? if so we should name is `user_identifier`

Reply:
`user_identifier` is a broad & subjective term. I will rename it to `username_or_email`.

Reply:
Sure.